### PR TITLE
PLT-132:Added support for properties in application profiles

### PR DESCRIPTION
**Pull request by SivaanandM**
Jira - https://spectrocloud.atlassian.net/browse/PLT-132

Terraform - Properties field missing for app profile type operator-instance

### CODE REVIEW

File: spectrocloud/resource_application_profile.go [hunk 1]
```diff
@@ -105,6 +105,13 @@ func resourceApplicationProfile() *schema.Resource {
 							Description: "The name of the specified pack.",
 							Required:    true,
 						},
+						"properties": {
```
Comment (karl-cardenas-coding) [inline spectrocloud/resource_application_profile.go:108]: @SivaanandM @nikchern we need to add the description fields to input attributes.

File: docs/resources/application_profile.md [hunk 2]
```diff
@@ -164,8 +164,8 @@ Optional:
 - `tag` (String) The identifier or version to label the pack.
 - `type` (String) The type of Pack. Allowed values are `container`, `helm`, `manifest`, or `operator-instance`.
 - `uid` (String) The unique id of the pack. This is a computed field and is not required to be set.
-- `values` (String) The values to be used for the pack. This is a stringified JSON object.
-
+- `values` (String) The values to be used for the pack. This is a string field JSON object.
+- `properties` (map) The key and value pair required for dependents packs
```
Comment (karl-cardenas-coding) [inline docs/resources/application_profile.md:168]: @SivaanandM This description belongs in the resource file. The command `go generate` populates all of the docs pages, so unless it's added in the description then this value will be overwritten the next time someone issues the command `go generate`

File: spectrocloud/resource_application_profile.go [hunk 3]
```diff
@@ -105,6 +105,13 @@ func resourceApplicationProfile() *schema.Resource {
...
+							Type:     schema.TypeMap,
```
Comment (karl-cardenas-coding) [inline spectrocloud/resource_application_profile.go:109]:
```suggestion
							Type:     schema.TypeMap,
							Description: "The key and value pair required for dependents packs."
```

I used the definition @SivaanandM created but when I read it I still don't get a good sense of what this is. Is there a better description we can use?

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Left some feedback related to documentation.

File: spectrocloud/resource_application_profile.go (hunk 1 above)
Comment (SivaanandM) [reply spectrocloud/resource_application_profile.go:108]: Added description

File: spectrocloud/resource_application_profile.go (hunk 3 above)
Comment (SivaanandM) [reply spectrocloud/resource_application_profile.go]: updated

**Review by karl-cardenas-coding — APPROVED** [review]
Thank you @SivaanandM looks good, just caught one minor grammar fix.

File: examples/resources/spectrocloud_application_profile/resource.tf [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  name = "profile-herbs"
+  version = "1.0.0"
+  context = "project"
+  tags = ["sivaa", "terraform"]
+  description = "test"
+  cloud = "all"
+  pack {
+    type = "operator-instance"
+    name = "mysql-3-stage"
+    source_app_tier="636c0714c196e565df7a7b37"
```
Comment (nikchern) [inline examples/resources/spectrocloud_application_profile/resource.tf:11]: could you please leave here a datasource example?

File: spectrocloud/resource_application_profile.go [hunk 5]
```diff
@@ -507,14 +537,26 @@ func toApplicationProfilePackUpdate(pSrc interface{}) *models.V1AppTierUpdateEnt
 		})
 	}
 
+	pProperties := make([]*models.V1AppTierPropertyEntity, 0)
```
Comment (nikchern) [inline spectrocloud/resource_application_profile.go:540]: reuse as function

File: spectrocloud/resource_application_profile.go [hunk 6]
```diff
@@ -393,6 +410,17 @@ func toApplicationProfilePackCreate(pSrc interface{}) (*models.V1AppTierEntity,
 	}
 	pType := models.V1AppTierType(p["type"].(string))
 
+	pProperties := make([]*models.V1AppTierPropertyEntity, 0)
```
Comment (nikchern) [inline spectrocloud/resource_application_profile.go:413]: reuse as function as below

**Pull request merged**